braindecode/models/modules.py

```diff
@@ -528,102 +529,289 @@ def extra_repr(self) -> str:
-        self.lowpass_filter_width = lowpass_filter_width
-        self.rolloff = rolloff
-        self.beta = beta
-
-        if self.orig_freq != self.new_freq:
-            kernel, self.width = _get_sinc_resample_kernel(
-                self.orig_freq,
-                self.new_freq,
-                self.gcd,
-                self.lowpass_filter_width,
-                self.rolloff,
-                self.resampling_method,
-                beta,
-                dtype=dtype,
+        # and we accepted as
+        self.band_filters = band_filters
+        self.n_bands = len(band_filters)
+        self.phase = phase
+        self.method = method
+        self.n_chans = n_chans
+
+        self.method_iir = True if self.method == "iir" else False
+
+        if self.method_iir:
+            if iir_params is None:
+                iir_params = dict(output="ba")
+            else:
+                if "output" in iir_params:
+                    if iir_params["output"] == "sos":
+                        warn("""
+                            It is not possible to use second-order section
+                            filtering with Torch. Changing to filter ``ba``
+                            """)
+                        iir_params["output"] = "ba"
+
+        filts = {}
+        for idx, (l_freq, h_freq) in enumerate(band_filters):
+            filt = create_filter(
+                data=None,
+                sfreq=sfreq,
+                l_freq=l_freq,
+                h_freq=h_freq,
+                filter_length=filter_length,
+                l_trans_bandwidth=l_trans_bandwidth,
+                h_trans_bandwidth=h_trans_bandwidth,
+                method=method,
+                iir_params=iir_params,
+                phase=phase,
+                fir_window=fir_window,
+                fir_design=fir_design,
+                verbose=verbose,
             )
-            self.register_buffer("kernel", kernel)
+            if not self.method_iir:
+                filts[f"{idx}_band"] = from_numpy(filt).float()
+            else:
+                b = from_numpy(filt["b"]).float()
+                a = from_numpy(filt["a"]).float()
+
+                filts[f"{idx}_band"] = (b, a)
 
-    def forward(self, waveform: Tensor) -> Tensor:
-        r"""
-        Args:
-            waveform (Tensor): Tensor of audio of dimension (..., time).
+            self.filts = filts
 
-        Returns:
-            Tensor: Output signal of dimension (..., time).
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
         """
-        if self.orig_freq == self.new_freq:
-            return waveform
-        return _apply_sinc_resample_kernel(
-            waveform, self.orig_freq, self.new_freq, self.gcd, self.kernel, self.width
-        )
+        Apply the filter bank to the input signal.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor of shape (batch_size, time_points).
```

Comment:
missing `n_chans`?

Reply:
Aren't you missing `n_chans` in the shape?
```suggestion
            Input tensor of shape (batch_size, n_chans, time_points).
```